### ISSUE TITLE
Navigation gets painfully slow after a while

### ISSUE THREAD

**Issue by BernhardBln** [42Crunch/intellij-openapi-editor#43]
We have an open-api file with around 5.000 lines.

When I start Intellij and open that file, I can quickly navigate around in the file, using either the structure view, using the breadcrumbs (like jumping to one of the parents) or using CMD+mouse click (I'm on Mac) on a reference.

This works very fast, as expected. 

However, when I keep working in that file for a while, the speed gradually degrades, first needs a short moment before jumping at the expected element, in the end needs like 4-5 seconds every time, until I restart IntelliJ. (It is not enough to close the editor tab of the open-api file).

Any ideas?

IntelliJ IDEA 2023.3.4 (Ultimate Edition)
Build #IU-233.14475.28, built on February 13, 2024

Runtime version: 17.0.10+1-b1087.17 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 14.3.1
GC: G1 Young Generation, G1 Old Generation
Memory: 6144M
Cores: 12
Metal Rendering is ON
Registry:
  ide.appIcon.requestAttention=false
  ide.tooltip.initialDelay=674
  debugger.new.tool.window.layout=true
  analyze.exceptions.on.the.fly=true
  ide.experimental.ui=true
Non-Bundled Plugins:
  be.jbeckers.compare_tab_with_editor2 (1.0.17)
  com.alexanderpa.flyway.migration.creator (2.1)
  com.mzyupc.a-redis (1.0.0)
  StringToolsPlugin (4.1)
  com.jetbrains.ide.streamdeck (2023.3.2)
  org.intellij.plugins.hcl (233.13135.65)
  com.intellij.plugins.watcher (233.13135.65)
  intellij-clock (2.0.0)
  AceJump (3.8.18)
  IdeaVIM (2.9.1)
  org.jetbrains.IdeaVim-EasyMotion (1.11)
  com.github.holgerbrandl.pasteimages/ (1.3.2)
  google-java-format (1.19.2.0)
  com.krrrr38.idea.mockito.postfix (0.0.9)
  VisualVMLauncher (1.23.1-IJ2023.3)
  IntelliTail (3.1.3)
  String Manipulation (9.12.0)
  com.sunny.plugin.MockitoGenPlugin (1.5.4)
  GrepConsole (12.25.0-IJ2023.3)
  PlantUML integration (7.7.0-IJ2023.2)
  MavenRunHelper (4.27.1-IJ2022.2)
  com.github.copilot (1.4.17.4734)
  com.crunch42.openapi (1.82)
  org.exbin.deltahex.intellij (0.2.9)
  com.intellij.lang.jsgraphql (233.13135.65)
  org.mapstruct.intellij (1.6.1)
  org.sonarlint.idea (10.3.0.77475)
  mobi.hsz.idea.gitignore (4.5.2)
  com.github.amolenaar.idea.clock (1.0)
  com.intellij.plugins.html.instantEditing (233.13135.65)
Kotlin: 233.14475.28-IJ

**Comment by vfateev** [comment]
Is it possible to share the file you are using?
Will it happen if you disable or remove the plugin?
How long did you work with the file to see 4-5 seconds of speed degradation?

**Comment by BernhardBln** [comment]
I unfortunately cannot share the file, it's from a client.

I think was maybe 10-15 minutes of work until it got slow.

Unfortunately I now finished with my refactoring of the file, and opening it now and trying to provoke by doing some random changes it does not work 🫠 However, this happened very consistently when I was working on it, and also on machines of colleagues.

I tried with a big one from the internet, that is publicly available and has 25.000 lines of codes, but works like a charm. Only difference to our file is that that one does not have any examples, while we have a lot.

My suggestion: I'll close this now (as I'm also off for vacation now), and next time I work with our file I will disable the plugin again if it happens again, and will reopen here to let you know if that made a difference.

Sorry I cannot give you better feedback, but maybe sb with a similar problem has a file they can share.
